inseq/attr/feat/perturbation_attribution.py

```diff
@@ -0,0 +1,74 @@
+from typing import Any, Dict
+
+import logging
+
+from captum.attr import Occlusion
+
+from ...data import PerturbationFeatureAttributionStepOutput
+from ...utils import Registry
+from ..attribution_decorators import set_hook, unset_hook
+from .attribution_utils import get_source_target_attributions
+from .gradient_attribution import FeatureAttribution
+
+
+logger = logging.getLogger(__name__)
+
+
+class PerturbationMethodRegistry(FeatureAttribution, Registry):
+    """Occlusion-based attribution methods."""
+
+    @set_hook
+    def hook(self, **kwargs):
+        pass
+
+    @unset_hook
+    def unhook(self, **kwargs):
+        pass
+
+
+class OcclusionAttribution(PerturbationMethodRegistry):
+    """Occlusion-based attribution method.
+    Reference implementation:
+    `https://captum.ai/api/occlusion.html <https://captum.ai/api/occlusion.html>`__.
+
+    Usages in other implementations:
+    `niuzaisheng/AttExplainer <https://github.com/niuzaisheng/AttExplainer/blob/main/baseline_methods/\
+    explain_baseline_captum.py>`__
+    `andrewPoulton/explainable-asag <https://github.com/andrewPoulton/explainable-asag/blob/main/explanation.py>`__
+    `copenlu/xai-benchmark <https://github.com/copenlu/xai-benchmark/blob/master/saliency_gen/\
+    interpret_grads_occ.py>`__
+    `DFKI-NLP/thermostat <https://github.com/DFKI-NLP/thermostat/blob/main/src/thermostat/explainers/occlusion.py>`__
+    """
+
+    method_name = "occlusion"
+
+    def __init__(self, attribution_model, **kwargs):
+        super().__init__(attribution_model)
+        self.is_layer_attribution = False
+        self.method = Occlusion(self.attribution_model)
+
+    def attribute_step(
+        self,
+        attribute_fn_main_args: Dict[str, Any],
+        attribution_args: Dict[str, Any] = {},
+    ) -> Any:
+
+        if "sliding_window_shapes" not in attribution_args:
+            # Sliding window shapes is defined as a tuple
+            # First entry is between 1 and length of input
+            # Second entry is given by the max length of the underlying model
```

Comment:
I'm a bit puzzled by the second entry: the max length you take here is the max generation length that the model can handle, but my understanding was that this would be the `hidden_size` from the model config, to ensure that there is no partial masking of token embeddings. Could you clarify?

Reply:
You're right, I accidentally took the next best attribute in `self.attribution_model` that had 512 as the size. This was careless.
The second entry should rather be based on the embedding size, right?
Does accessing it via `self.attribution_model.get_embedding_layer()` make sense?